internal/providers/pluginfw/resources/sharing/data_share.go

```diff
@@ -0,0 +1,74 @@
+package sharing
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go/service/sharing"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/sharing_tf"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+)
+
+func DataSourceShare() datasource.DataSource {
+	return &ShareDataSource{}
+}
+
+var _ datasource.DataSourceWithConfigure = &ShareDataSource{}
+
+type ShareDataSource struct {
+	Client *common.DatabricksClient
+}
+
+func (d *ShareDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = pluginfwcommon.GetDatabricksStagingName("share")
```

Comment:
nit: We should abstract this out above like: https://github.com/databricks/terraform-provider-databricks/blob/main/internal/providers/pluginfw/resources/volume/data_volumes.go#L19

Reply:
See my comment below.